File: evidence/generated-systems/angular/healthcare-on-angular/src/app/models/CareTask.js

// Define collection and schema for CareTask
export interface CareTask {
    description:
	type : string
    dueDate:
	type : Date
    CarePlan:
	type : Schema.Types.ObjectId
    AssignedTo:
	type : Schema.Types.ObjectId
    Encounter:
	type : Schema.Types.ObjectId
    Status:
 	type : String
    Priority:
 	type : String
#
    collection: 'careTasks'
}

File: evidence/generated-systems/rubyonrails/healthcare-on-rails/healthcareonrails/app/models/caretask.rb
class CareTask < ApplicationRecord
  enum Status: [:Requested, :Accepted, :InProgress, :Completed, :Cancelled, :Failed]
  enum Priority: [:Routine, :Urgent, :Stat]


  has_many :CarePlan, class_name: 'CarePlan'
  has_many :AssignedTo, class_name: 'Clinician'
  has_many :Encounter, class_name: 'Encounter'

end
